### DISCUSSION TITLE
How to Monitor PLC Connection State

### DISCUSSION THREAD

**Discussion by CGuiho** [SerafinTech/ST-node-ethernet-ip#70] [Q&A]
Hi,

First, I'd like to extend my gratitude for creating and maintaining this library. It's a fantastic tool, and I've been using it for about six weeks. 

However, I've encountered a challenge that I hope to get some guidance on. Specifically, I'm looking for effective strategies to **monitor the connection state of a PLC**. In my application, it's crucial to be immediately aware if the PLC loses its connection.

Is there a built-in method or recommended approach to watch for connection state changes? Ideally, I would like to subscribe to connection events or set up a callback function that triggers when the PLC disconnects. 

Thank you in advance for your assistance. Any suggestions or examples would be greatly appreciated!

Best regards,

Christ

**Comment by CGuiho** [comment]
I've already tried some approach, such as:
- Reading tag list (where I get timeout and crushs the app even though I've put it around `try and catch`.
- Reading properties (which are always available after first connection).

**Comment by xszong** [comment]
链接时候的握手信号不行嘛？ plc和plc直接加个心跳信号
